includes/class-swedbank-pay-payment-gateway-checkout.php

```diff
@@ -554,19 +566,27 @@ public function return_handler() {
 				throw new \Exception( 'Error: Invalid transaction number' );
 			}
 
-			// Create Background Process Task
-			$background_process = new Swedbank_Pay_Background_Queue();
-			$background_process->push_to_queue(
+			// Schedule the payment for later processing.
+			$schedule_id = as_schedule_single_action(
+				time() + 30,
+				Swedbank_Pay_Scheduler::ACTION_ID,
 				array(
 					'payment_method_id' => $this->id,
 					'webhook_data'      => $raw_body,
 				)
 			);
```

Comment:
Question (blocking): I know this is currently in the code, but is the plugin relying on the body as it is sent to the callback for a means of verifying the order? That seems like a potential risk that could be open to exploit. We should probably in the scheduler make a request to read the order from Swedbank instead to ensure the data is real and has not been sent by a separate entity.

Reply:
There is verification happening [here](https://github.com/krokedil/swedbank-pay-woocommerce-paymentmenu/blob/main-action-scheduler/includes/class-swedbank-pay-scheduler.php#L136).

Reply:
@mntzrr That links to a todo comment, so is this something yet to be implemented or do you mean the line bellow when we finalize the payment? I do see us reading the order in the finalize method though, so it should be ok then! Thank you for checking!